src/autodiscovery.ts

```diff
@@ -256,10 +266,56 @@ export class AutoDiscovery {
             }
         });
 
+        const authConfig = await this.validateDiscoveryAuthenticationConfig(wellknown);
+        clientConfig[M_AUTHENTICATION.stable!] = authConfig;
+
         // Step 8: Give the config to the caller (finally)
         return Promise.resolve(clientConfig);
     }
 
+    public static async validateDiscoveryAuthenticationConfig(
```

Comment:
Can we add a JSDoc explaining what this does?
It seems to be doing a fair few things and there's a few failure scenarios that are not obvious